codec-http/src/main/java/io/netty/handler/codec/http/HttpMethod.java

```diff
@@ -173,4 +173,44 @@ public String toString() {
     public int compareTo(HttpMethod o) {
         return name().compareTo(o.name());
     }
+
+    private static final class EnumNameMap<T> {
+        private final EnumNameMap.Node<T>[] values;
+        private final int valuesMask;
+
+        EnumNameMap(EnumNameMap.Node<T>... nodes) {
+            values = (EnumNameMap.Node<T>[]) new EnumNameMap.Node[findNextPositivePowerOfTwo(nodes.length)];
+            valuesMask = values.length - 1;
+            for (EnumNameMap.Node<T> node : nodes) {
+                int i = hashCode(node.key) & valuesMask;
+                if (values[i] != null) {
+                    throw new IllegalArgumentException("index " + i + " collision between values: [" +
+                            values[i].key + ", " + node.key + "]");
+                }
+                values[i] = node;
+            }
+        }
+
+        T get(String name) {
+            EnumNameMap.Node<T> node = values[hashCode(name) & valuesMask];
+            return node == null || !node.key.equals(name) ? null : node.value;
+        }
+
+        private static int hashCode(String name) {
+            // This hash code needs to produce a unique index for each HttpMethod. If new methods are added this
+            // algorithm will need to be adjusted. The goal is to have each enum name's hash value correlate to a unique
+            // index in the values array.
+            return name.hashCode() >>> 6;
+        }
+
+        private static final class Node<T> {
```

Comment:
This one is exposed via a package-private constructor so it should be package-private as well.

Reply:
`private` inner class with `package private` constructor is common practice. Can you clarify why this needs to change? The inner class is private (only visible to the outer class), but the constructor is `package private` because its internal to the class and the outer class needs to construct it.

Reply:
I read at some point that the JVM does some extra steps while dispatch stuff in this case while it not does when it is package-private or public. Maybe this is not true anymore.

Reply:
never mind... I somehow thought the EnumNameMap is package-private ...